src/lib/comms/sol-lwm2m-client.c

```diff
@@ -2131,11 +2181,12 @@ server_connection_ctx_new(struct sol_lwm2m_client *client,
     SOL_INT_CHECK_GOTO(r, < 0, err_append);
     conn_ctx->client = client;
     conn_ctx->server_id = server_id;
+    conn_ctx->secure = secure;
     sol_vector_init(&conn_ctx->server_addr_list,
         sizeof(struct sol_network_link_addr));
 
     if (!uri.port)
-        conn_ctx->port = SOL_LWM2M_DEFAULT_SERVER_PORT;
```

Comment:
Nice catch, I overlooked this function.
Fixed:
```C
    if (!uri.port)
        if (secure)
            conn_ctx->port = SOL_LWM2M_DEFAULT_SERVER_PORT_DTLS;
        else
            conn_ctx->port = SOL_LWM2M_DEFAULT_SERVER_PORT_COAP;
    else
        conn_ctx->port = uri.port;
```

Also, I've added an URI Scheme enforcement, right before the `calloc` call:
```C
    SOL_NULL_CHECK(&uri.scheme, NULL);
    if (sol_str_slice_str_case_eq(uri.scheme, "coaps"))
        SOL_EXP_CHECK(!secure, NULL);
    else if (sol_str_slice_str_case_eq(uri.scheme, "coap"))
        SOL_EXP_CHECK(secure, NULL);
    else
        return NULL;
```